### DISCUSSION TITLE
How can I use different languages in email templates (text and subject)

### DISCUSSION THREAD

**Discussion by ludi81** [directus/directus#8236] [Archive]
Hi,

we use directus with an app that supports multiple languages. If a user requests a new password (forgot password) at the moment the system returns an english email (password-reset.liquid file).
It would be great if the system sends the email in the right language depending on the Accept-Language header.

Do I really need to create a custom endpoint?
Thanks

**Comment by joselcvarela** [comment]
Your only solution is to implement custom email templates and maybe a custom endpoint or custom hook too.
https://docs.directus.io/guides/email-templates/?#email-templates

We do not have and do not intend (am I right @rijkvanzanten) to translate email templates since they should vary a lot for each use case.

**Reply by rijkvanzanten** [reply]
At this point in time, we don't have the email templates translated, that's correct. You can currently _overwrite_ the email template by adding a custom template with the same name (for example `password-reset.liquid`), though that won't solve for the language-per-user problem.

**Reply by ludi81** [reply]
An easy solution would be if you can provide the language parameter (user language or Accept-Language) for the template (liquid). Then we can use an if condition to render different html code.
`{% if lang == 'de' %}
    DE
{%  else %}
    EN
{% endif %}` 

or better: 
e.g. if the user prefers the language DE the system looks for the file e.g. password-reset-DE.liquid. If the file does not exist, the default file password-reset.liquid will be used.

**Reply by rijkvanzanten** [reply]
I was wondering whether we want to recreate the whole template, or offer a way to use a translations key (just like the app) inside the template. One way could be by creating a custom "translate" filter in liquid, so you could do something like:

```liquid
<p>{{ "password_reset_intro" | translate }}</p>
```

**Comment by rijkvanzanten** [comment]
I've opened this new feature request we can use to track this functionality: https://github.com/directus/directus/discussions/8239
